dialogflow/entity_management.rb

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def list_entities project_id:, entity_type_id:
+  # [START dialogflow_list_entities]
+  # project_id = "Your Google Cloud project ID"
+  # entity_type_id = "Existing Entity Type ID"
+  
+  require "google/cloud/dialogflow"
+
+  entity_types_client = Google::Cloud::Dialogflow::EntityTypes.new
+  parent = entity_types_client.class.entity_type_path project_id, entity_type_id
+
+  entities = entity_types_client.get_entity_type(parent).entities
+
+  entities.each do |entity|
+    puts "Entity value:    #{entity.value}"
+    puts "Entity synonyms: #{entity.synonyms}"
+  end
+  # [END dialogflow_list_entities]
+end
+
+
+def create_entity project_id:, entity_type_id:, entity_value:, synonyms:
+  # [START dialogflow_create_entity]
+  # project_id = "Your Google Cloud project ID"
+  # entity_type_id = "Existing Entity Type ID"
+  # entity_value = "New Entity Value"
+  # synonyms = ["synonym1", "synonym2"]
+  
+  require "google/cloud/dialogflow"
+
+  entity_types_client = Google::Cloud::Dialogflow::EntityTypes.new
+  entity_type_path = entity_types_client.class.entity_type_path project_id, entity_type_id
+
+  entity = { value: entity_value, synonyms: synonyms }
+
+  response = entity_types_client.batch_create_entities entity_type_path, [entity]
+
+  puts "Entity created: #{response}"
```

Comment:
What does this print out?

As this is a batch rpc method, I'm guessing that it returns multiple response objects, 1 per entity?

*Ah! Nope!*

`BatchCreateEntities` returns an Operation, it's a LRO!

https://cloud.google.com/dialogflow-enterprise/docs/reference/rpc/google.cloud.dialogflow.v2beta1#google.cloud.dialogflow.v2beta1.EntityTypes.BatchCreateEntities-table

So this sample should probably follow the usual template for LRO samples (?):
 - print out operation info
 - wait for operation to complete
 - print that it completed

Reply:
The op returns Empty, so there's no .result field on it to print out, but the existing code only *kicks off the LRO*, there's no knowing that it completed

Reply:
done.

the operation has at least the `done?` method.